public/app/features/alerting/unified/components/rule-editor/alert-rule-form/simplifiedRouting/SimplifiedRouting.tsx

```diff
@@ -0,0 +1,230 @@
+    const existingContactPoint = state.find((cp) => cp.alertManager.name === alertManager.name);
+
+    if (existingContactPoint) {
+      existingContactPoint.selectedContactPoint = receiver;
+    } else {
+      state.push(newContactPoint);
+    }
+  });
+});
+
+export function SimplifiedRouting() {
+  const { getValues, setValue } = useFormContext<RuleFormValues & { location?: string }>();
+  const styles = useStyles2(getStyles);
+  const contactPointsInAlert = getValues('contactPoints');
+
+  const alertManagerMetaData = useGetAlertManagersMetadata();
+
+  const alertManagerMetaDataPostable = alertManagerMetaData.filter((am) => am.postable);
+
+  // we merge the selected contact points with the alert manager meta data
+  const alertManagersWithSelectedContactPoints = alertManagerMetaDataPostable.map((am) => {
+    const selectedContactPoint = contactPointsInAlert?.find((cp) => cp.alertManager === am.name);
+    return { alertManager: am, selectedContactPoint: selectedContactPoint?.selectedContactPoint };
+  });
+
+  // use reducer to keep this alertManagersWithSelectedContactPoints in the state
+  const [alertManagersWithCPState, dispatch] = useReducer(receiversReducer, alertManagersWithSelectedContactPoints);
+
+  function getContactPointsForForm(alertManagersWithCP: AMContactPoint[]) {
+    return alertManagersWithCP.map((am) => {
+      return { alertManager: am.alertManager.name, selectedContactPoint: am.selectedContactPoint };
+    });
+  }
+
+  // whenever we update the receiversState we have to update the form too
+  useEffect(() => {
+    const contactPointsForForm = getContactPointsForForm(alertManagersWithCPState);
+    setValue('contactPoints', contactPointsForForm, { shouldValidate: false });
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [alertManagersWithCPState]);
+
+  //todo: decide what to do when there some alert managers not postable
+  //const shouldShowAM = alertManagerMetaDataPostable.length > 1;
+  const shouldShowAM = true;
+
+  return alertManagersWithCPState.map((alertManagerContactPoint, index) => {
+    return (
+      <div key={index}>
+        <Stack direction="column">
+          {shouldShowAM && (
+            <Stack direction="row" alignItems="center">
+              <div className={styles.firstAlertManagerLine}></div>
+              <div className={styles.alertManagerName}>
+                {' '}
+                Alert manager:
+                <img src={alertManagerContactPoint.alertManager.img} alt="" className={styles.img} />
+                {alertManagerContactPoint.alertManager.name}
+              </div>
+              <div className={styles.secondAlertManagerLine}></div>
+            </Stack>
+          )}
+          <ContactPointSelector
+            selectedReceiver={alertManagerContactPoint.selectedContactPoint}
+            dispatch={dispatch}
+            alertManager={alertManagerContactPoint.alertManager}
+          />
+        </Stack>
+      </div>
+    );
+  });
+}
+
```

Comment:
there was the `useReceiversMetadata` function that returned the same Map,  but the keys were the `Receiver` and in this case I need to get by receiver's `name`. Maybe we could refactor the other method and add a parameter to be able to get by name?